### ISSUE TITLE
the loss funtion here is the IoU loss in the paper?

### ISSUE THREAD

**Issue by luciaL** [0bserver07/One-Hundred-Layers-Tiramisu#12]

**Comment by szymko1995** [comment]
`model.compile(... , sample_weight_mode="temporal")`
You can calculate class weights with sklearn.utils.class_weight.compute_class_weight. Just make sure to flatten N-dim array.
Then if you have 4 classes you should create this kind of matrix:
```
class_weights = np.zeros((224,224, 4))
class_weights[:,:, 0] += 0.25
class_weights[:,:, 1] += 4
class_weights[:,:, 2] += 2
class_weights[:,:, 3] += 6
```
And finally:
`model.fit(... , class_weight=class_weights)`

Note, that it has been taken from keras's github. Works for me.

**Comment by luciaL** [comment]
Thank you so much.I will try it.

**Comment by szymko1995** [comment]
Okay. So, when you have 4 classes, that you want to segment output matrix of NN will be WxHxN_classes.
You set calculated weights (by for example sklearn.utils.class_weight.compute_class_weight) and paste them into each of 3-axis's WxH matrices. This way of setting weights for segmentation NN has been proposed by someone on StackOverflow or keras' GitHub as the output of segCNN isn't typical.
Idk if API in Keras for this kind of NN has changed over a time, but I think it should work.
